src/Interpreters/Context.h

```diff
@@ -282,6 +284,12 @@ class Context: public std::enable_shared_from_this<Context>
     /// A flag, used to distinguish between user query and internal query to a database engine (MaterializePostgreSQL).
     bool is_internal_query = false;
 
+    /// Background executors for *MergeTree tables
+    /// Must be in global context
+    MergeTreeBackgroundExecutorPtr merge_mutate_executor;
```

Comment:
Maybe move it to shared part?
https://github.com/ClickHouse/ClickHouse/blob/750e35a4546ed11b49b38b439d0443c5ee89244d/src/Interpreters/Context.cpp#L304